samples/dnn/audio_visual_speech_recognition.py

```diff
@@ -0,0 +1,265 @@
+        self.model.setInput(video, 'video_input')
+        self.model.setInput(audio, 'audio_input')
+        out = self.model.forward()
+        pred = out[0].argmax()
+        pred = self.labels[pred] if self.labels else pred
+        return pred
+
+    def run(self):
+        '''
+        Read the video and process it.
+        '''
+        self.cap.open(self.source, cv2.CAP_MSMF, self.params)
+        if not self.cap.isOpened():
+            print('Cannot open video source')
+            exit(1)
+        audioBaseIndex = int(self.cap.get(cv2.CAP_PROP_AUDIO_BASE_INDEX))
+        audioChannels = int(self.cap.get(cv2.CAP_PROP_AUDIO_TOTAL_CHANNELS))
+
+        if self.cap.isOpened():
+            while self.cap.grab():
+                ret, frame = self.cap.retrieve()
+                frame = cv2.resize(frame, (self.width, self.height))
+                audioFrame = np.asarray([])
+                audioFrame = self.cap.retrieve(audioFrame, audioBaseIndex)
+                audioFrame = audioFrame[1][0] if audioFrame is not None else None
+
+                if self.realtime:
+                    img, aud = self.preprocess(frame, audioFrame)
+                    if img is not None and aud is not None:
+                        img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+                        img = cv2.resize(img, (96,96))
+                        self.frames_queue.append(img)
+                        for i in range(len(aud)):
+                            self.audio_queue.append(aud[i])
+                        if len(self.frames_queue) < self.margin:
+                            continue
+                        pred = self.predict(self.frames_queue, self.audio_queue)
+                        print(pred)
+                        if self.show_video:
+                            cv2.circle(frame, np.mean(self.landmarks_queue, axis=0)[0][5].astype(np.int32), 2, (0,0,255), -1)
+                            cv2.circle(frame, np.mean(self.landmarks_queue, axis=0)[0][6].astype(np.int32), 2, (0,0,255), -1)
+                            cv2.imshow('frame', frame)
+                            if cv2.waitKey(1) & 0xFF == ord('q'):
+                                break
+                else:
+                    self.frames_queue.append(frame)
+                    self.audio_queue.append(audioFrame)
+        if not self.realtime:
+            image_queue = []
+            audio_queue = []
+            for i in range(len(self.frames_queue)):
+                img, aud = self.preprocess(self.frames_queue[i], self.audio_queue[i])
+                if img is not None and aud is not None:
+                    img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+                    img = cv2.resize(img, (96,96))
+                    image_queue.append(img)
+                    for i in range(len(aud)):
+                        audio_queue.append(aud[i])
```

Comment:
Conflict with `i` defined above
```suggestion
                    for audio in aud:
                        audio_queue.append(audio)
```